pkg/generator/generator.go

```diff
@@ -29,23 +30,24 @@ var (
 type KeyMgr interface {
 	References() ([]string, []string)
 	LoadReferenceData(data map[string][]byte) error
-	LoadSecretFromManager(context context.Context, config *v1alpha1.AppConfig, namespace, secretName string) error
-	EnsureSecretManager(context context.Context, config *v1alpha1.AppConfig, namespace, secretName string) error
+	LoadSecretFromManager(context context.Context, sm secretsmanager.SecretManager, namespace, secretName string) error
```

Comment:
SecretManager should be available in `GenConfig` you shouldn't need to pass it as an arg

Reply:
ignore. I don't think this is correct.